supplemental_notebooks/2021-07-01_muddiest_points.Rmd

```diff
@@ -0,0 +1,269 @@
+
+Let's look at this in a little more depth by generating random gene sets of different sizes and examining the enrichment scores that we get back out of `GSEA()`.
+
+```{r}
+# Create a vector of all gene symbols
+all_genes <- markers_df$gene_symbol
+
+# Make a list of integers that indicate the random gene set sizes
+gene_set_size <- list(15, 25, 50, 100, 250, 500)
+
+# Set number of replicates
+nreps <- 1000
+# Generate 1000 random gene sets of each size
+random_gene_sets <- rep(gene_set_size, nreps) %>%  # Repeat gene sizes so we run `nreps` times 
+  purrr::map(
+    # Sample the vector of all genes, choosing the number of items specified
+    # in the element of gene set size
+    ~ base::sample(x = all_genes,
+                   size = .x)
+  )
+
+# We will include the size of the gene set in the gene set name 
+# Start by taking the length of each pathway and appending "pathway_" to that
+# number
+lengths_vector <- random_gene_sets %>%
+  # Get the length of each gene set (number of genes)
+  purrr::map(~ length(.x)) %>%
+  # Make it "pathway_<gene set size>"
+  purrr::map(~ paste0("pathway_", .x)) %>%
+  # Return a vector
+  purrr::flatten_chr()
+
+# Add the names in lengths_vector to the list - "pathway_<gene set size>"
+random_gene_sets <- random_gene_sets %>% 
+  # make.names() appends a "version" if something is not unique
+  purrr::set_names(nm = make.names(lengths_vector, unique = TRUE))
+
+# Make this a data frame that GSEA() can use
+random_gene_set_df <- reshape2::melt(random_gene_sets) %>%
+  dplyr::select(gene_set = L1, symbol = value)
+```
+
+Now perform GSEA using these random gene sets of different sizes.
+
+```{r}
+gsea_random <- GSEA(geneList = lfc_vector,  # ordered ranked gene list
+                     minGSSize = 15,  # minimum gene set size
+                     maxGSSize = 500,  # maximum gene set set
+                     pvalueCutoff = 1,  # Return everything!
+                     pAdjustMethod = "BH",  # correction for multiple hypothesis testing
+                     TERM2GENE = random_gene_set_df)  # data frame with randomly generated gene sets
+```
+Let's get the enrichment scores and normalized enrichment scores in a data frame we'll use for plotting.
+
+```{r}
+random_es <- gsea_random@result %>%
+  # Just the pathway name, enrichment score, and normalized enrichment score
+  dplyr::select(ID, enrichmentScore, NES) %>%
+  # Create a new column that includes the gene set size as a factor in ascending
+  # order
+  dplyr::mutate(pathway_size = factor(
+    stringr::str_remove(stringr::str_remove(ID ,"\\..*"), ".*\\_" ),
+    levels = c(15, 25, 50, 100, 250, 500)
+  ))
+
+# Take a peek
+head(random_es)
+```
+
+Let's create a ridgeline plot of enrichment scores.
+
+```{r}
+random_es %>%
+  ggplot2::ggplot(ggplot2::aes(enrichmentScore,
+                               pathway_size,
+                               fill = pathway_size)) +
+  ggridges::geom_density_ridges() +
```

Comment:
We don't have `ggridges`! We can show how to install it here (or a bit earlier) if we like.